uvicorn/middleware/proxy_headers.py

```diff
@@ -14,11 +14,21 @@ class ProxyHeadersMiddleware:
     def __init__(self, app, trusted_hosts="127.0.0.1"):
         self.app = app
         if isinstance(trusted_hosts, str):
-            self.trusted_hosts = [item.strip() for item in trusted_hosts.split(",")]
+            self.trusted_hosts = {item.strip() for item in trusted_hosts.split(",")}
         else:
-            self.trusted_hosts = trusted_hosts
+            self.trusted_hosts = set(trusted_hosts)
         self.always_trust = "*" in self.trusted_hosts
 
+    def get_trusted_client_host(
+        self, x_forwarded_for_hosts
+    ):  # type: (List[str]) -> str
+        if self.always_trust:
+            return x_forwarded_for_hosts[0]
+
+        for host in reversed(x_forwarded_for_hosts):
+            if host not in self.trusted_hosts:
+                return host
```

Comment:
Do we need a fall through case so there's no way we can end up returning `None` here?

Reply:
I fail to see how it would be possible

Reply:
It would happen if all the IPs in the chain were explicitly listed in `trusted_hosts` - but this would only occur if `trusted_hosts` were incorrectly set and/or the first reverse proxy in the chain failed to set `x-forwarded-for` to the client's IP (not our fault).

In this case, is there any other justifiable return value than `None`? I can't think of one